frontend/src/views/Announcements/AnnouncementsList.tsx

```diff
@@ -0,0 +1,213 @@
+    useState<DtoAnnouncement>();
+  const [isCreateAnnouncementModalOpen, setIsCreateAnnouncementModalOpen] =
+    useState(false);
+  const [isEditAnnouncementModalOpen, setIsEditAnnouncementModalOpen] =
+    useState(false);
+
+  const handleDelete = (id: number) => {
+    openDeleteModal();
+    setAnnouncementToBeDeleted(id);
+  };
+
+  const deleteAnnouncement = async () => {
+    setDeleteModalLoading(true);
+    const announcementsApiUrl = `/api/announcements/${announcementToBeDeleted}`;
+    const response = await makeRequest<AnnouncementDeleteResponseDto>(
+      announcementsApiUrl,
+      'DELETE',
+    );
+    setDeleteModalLoading(false);
+    if (!response.ok) {
+      showErrorSnackbar();
+      return;
+    }
+    setDeleteModalLoading(false);
+    closeDeleteModal();
+    await mutate();
+    showSuccessSnackbar('Ogłoszenie zostało pomyślnie usunięte');
+  };
+
+  const handleCreate = () => {
+    setIsCreateAnnouncementModalOpen(true);
+  };
+
+  const createAnnouncement = async (values: AnnouncementFormSchema) => {
+    const { subject, body } = values;
+    const announcement: AnnouncementCreateRequestDto = {
+      announcement: {
+        subject,
+        body,
+      },
+    };
+    const announcementApiUrl = '/api/announcements';
+    const response = await makeRequest<
+      AnnouncementCreateResponseDto,
+      AnnouncementCreateRequestDto
+    >(announcementApiUrl, 'POST', announcement);
+    if (!response.ok) {
+      showErrorSnackbar();
+    }
+    showSuccessSnackbar('Ogłoszenie zostało utworzone');
+    setIsCreateAnnouncementModalOpen(false);
+    await mutate();
+  };
+
+  const handleEdit = (announcement: DtoAnnouncement) => {
+    setAnnouncementToBeEdited(announcement);
+    setIsEditAnnouncementModalOpen(true);
+  };
+
+  const editAnnouncement = async (values: AnnouncementFormSchema) => {
+    const { subject, body } = values;
+    const announcement: AnnouncementUpdateRequestDto = {
+      announcement: {
+        subject,
+        body,
+      },
+    };
+    const announcementApiUrl = `/api/announcements/${announcementToBeEdited?.id}`;
+    const response = await makeRequest<
+      AnnouncementUpdateResponseDto,
+      AnnouncementUpdateRequestDto
+    >(announcementApiUrl, 'PATCH', announcement);
+    if (!response.ok) {
+      showErrorSnackbar();
+    }
+    showSuccessSnackbar('Ogłoszenie zostało zapisane');
+    setIsEditAnnouncementModalOpen(false);
+    await mutate();
+  };
+
+  if (announcementsError) {
+    return <GeneralAPIError />;
+  }
+
+  if (announcementsData === undefined) {
+    return <FullPageLoading />;
+  }
+
+  const { announcements } = announcementsData;
+
+  return (
+    <Flex flexDirection="column" height="100%">
+      <Toolbar
+        sx={{
+          pl: { sm: 2 },
+          pr: { xs: 1, sm: 1 },
+          mb: '2rem',
+        }}
+      >
+        <Typography sx={{ flex: '1 1 30%' }} variant="h6" component="h1">
```

Comment:
Why do we need the `30%`?

Reply:
Don't exactly know where did it come from - I'll be changing this here and in instructors list to match the trainees list